rbac/default-role-manager/role_manager.go

```diff
@@ -16,19 +16,18 @@ package defaultrolemanager
 
 import (
 	"errors"
-	"strings"
 	"sync"
 
 	"github.com/casbin/casbin/log"
 	"github.com/casbin/casbin/rbac"
-	"github.com/casbin/casbin/util"
 )
 
 // RoleManager provides a default implementation for the RoleManager interface
 type RoleManager struct {
 	allRoles          *sync.Map
 	maxHierarchyLevel int
 	hasPattern        bool
+	matchingFunc      map[string]func(arg1, arg2 string) bool
```

Comment:
You can define ``map[string]func(arg1, arg2 string) bool`` to a type as the example used by our evaluator:
```go
type Function func(args ...interface{}) (interface{}, error)
```